examples/GPU/example_learn_samples_multicoil.py

```diff
@@ -0,0 +1,221 @@
+    axs[2].imshow(np.abs(recon[0][0].detach().cpu().numpy()), cmap="gray")
+    axs[2].axis("off")
+    axs[2].set_title("Reconstruction")
+    if loss is not None:
+        axs[3].plot(loss)
+        axs[3].set_title("Loss")
+        axs[3].grid("on")
+    if save_name is not None:
+        plt.savefig(save_name, bbox_inches="tight")
+        plt.close()
+    else:
+        plt.show()
+
+
+# %%
+# Setup model and optimizer
+# -------------------------
+n_coils = 6
+init_traj = initialize_2D_radial(32, 256).astype(np.float32).reshape(-1, 2)
+model = Model(init_traj, n_coils=n_coils, img_size=(256, 256))
+optimizer = torch.optim.Adam(model.parameters(), lr=1e-3)
+schedulder = torch.optim.lr_scheduler.LinearLR(
+    optimizer,
+    start_factor=1,
+    end_factor=0.1,
+    total_iters=100,
+)
+# %%
+# Setup data
+# ----------
+mri_2D = torch.from_numpy(np.flipud(bwdl.get_mri(4, "T1")[80, ...]).astype(np.float32))
+mri_2D = mri_2D / torch.mean(mri_2D)
+smaps_simulated = torch.from_numpy(birdcage_maps((n_coils, *mri_2D.shape)))
+mcmri_2D = mri_2D[None].to(torch.complex64) * smaps_simulated
+model.eval()
+recon = model(mcmri_2D)
+fig, axs = plt.subplots(1, 3, figsize=(15, 5))
+plot_state(axs, mri_2D, model.trajectory.detach().cpu().numpy(), recon)
+
+# %%
+# Start training loop
+# -------------------
+losses = []
+image_files = []
+model.train()
+
+with tqdm(range(20), unit="steps") as tqdms:
+    for i in tqdms:
+        out = model(mcmri_2D)
+        loss = torch.nn.functional.mse_loss(out, mri_2D[None, None])
+        numpy_loss = loss.detach().cpu().numpy()
+        tqdms.set_postfix({"loss": numpy_loss})
+        losses.append(numpy_loss)
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+        schedulder.step()
+        with torch.no_grad():
+            # Clamp the value of trajectory between [-0.5, 0.5]
+            for param in model.parameters():
+                param.clamp_(-0.5, 0.5)
+        # Generate images for gif
+        hashed = joblib.hash((i, "learn_traj", time.time()))
+        filename = "/tmp/" + f"{hashed}.png"
+        plt.clf()
+        fig, axs = plt.subplots(2, 2, figsize=(10, 10))
+        plot_state(
+            axs,
+            mri_2D,
+            model.trajectory.detach().cpu().numpy(),
+            out,
+            losses,
+            save_name=filename,
+        )
+        image_files.append(filename)
+
+
+# Make a GIF of all images.
+imgs = [Image.open(img) for img in image_files]
+imgs[0].save(
+    "mrinufft_learn_traj_mc.gif",
+    save_all=True,
+    append_images=imgs[1:],
+    optimize=False,
+    duration=2,
+    loop=0,
+)
+# sphinx_gallery_start_ignore
+# cleanup
+import os
+import shutil
+from pathlib import Path
+
+for f in image_files:
+    try:
+        os.remove(f)
+    except OSError:
+        continue
+# don't raise errors from pytest. This will only be executed for the sphinx gallery stuff
+try:
+    final_dir = (
+        Path(os.getcwd()).parent.parent
```

Comment:
you probably want to use `__file__` instead. You don't have the guarantee that you are running from the example directory.